pandas/core/nanops.py

```diff
@@ -421,7 +421,11 @@ def nanany(values, axis=None, skipna: bool = True, mask=None):
     >>> nanops.nanany(s)
     False
     """
-    values, _, _, _, _ = _get_values(values, skipna, fill_value=False, mask=mask)
+    values, _, dtype, _, _ = _get_values(values, skipna, fill_value=False, mask=mask)
+
+    if is_object_dtype(dtype) and axis is None:
```

Comment:
can you add a ref to this issue here (and in nanall where you added code): https://github.com/numpy/numpy/issues/4352, and a small comment ( like we are doing the right thing as numpy doesn't handle any/all on object arrays correct)